staging/src/k8s.io/apiserver/plugin/pkg/authenticator/token/webhook/webhook.go

```diff
@@ -69,7 +70,7 @@ func newWithBackoff(tokenReview authenticationclient.TokenReviewInterface, ttl,
 }
 
 // AuthenticateToken implements the authenticator.Token interface.
-func (w *WebhookTokenAuthenticator) AuthenticateToken(token string) (user.Info, bool, error) {
+func (w *WebhookTokenAuthenticator) AuthenticateToken(ctx context.Context, token string) (*authenticator.Response, bool, error) {
```

Comment:
TODOs here:
* take implicit audiences at WebhookTokenAuthenticator construction time
* if ctx is audience-limited, add into the tokenreview object
* if the tokenreview returns with an audience set that matches the ctx, copy into the response and return success
* if the tokenreview returns without an audience set, ensure the ctx audiences intersect with the implicit audiences taken at WebhookTokenAuthenticator construction time, and pick one to add into the response